### DISCUSSION TITLE
Setting up the CoG

### DISCUSSION THREAD

**Discussion by francesco1788** [capytaine/capytaine#105] [Q&A] [answered]
Good day,
when running an analysis I get the below message:

"_WARNING : The rotation dof Roll has been initialized around the origin of the domain (0, 0, 0).
WARNING : The rotation dof Pitch has been initialized around the origin of the domain (0, 0, 0).
WARNING : The rotation dof Yaw has been initialized around the origin of the domain (0, 0, 0)."

My understanding is that z=0, must always be at the waterline level. So -according to the warnings- the rotation will be around a point at the waterline. However, I want the RAO displacement associated to the CoG, so I want that the rotation DoF refers to the CoG and not to the Origin.

Could you tell me if I can set the rotation dofs to be about my CoG (assuming I know its coordinates)?

Thanks a lot!

**Comment by mancellin** [comment, marked as answer]
This is indeed poorly documented. There are two main ways to do that:

1. You can define rotation degrees of freedom around any axis in the following way:
```python
body.add_rotation_dof(
    axis=cpt.Axis(point=cog, vector=(1, 0, 0)),
    name="Roll",
)
```

2. There is an undocumented feature (that may not work anymore in future versions of the code): if the body has an attribute named `center_of_mass`, then it will be used to initialize the rigid body dofs
```python
body.center_of_mass = np.array([0, 1, -0.5])
body.add_all_rigid_body_dofs()
# INFO:capytaine.bodies.bodies:The rotation dof Roll has been initialized around the point: body.center_of_mass = [ 0.   1.  -0.5]

```

3. Finally, if you don't remember any of the above, there is always the tricky way :)
```python
body.translate(-cog)
# Center of mass is now (0, 0, 0)
body.add_all_rigid_body_dofs()
body.translate(cog)
```
